scout/load/panelapp.py

```diff
@@ -0,0 +1,132 @@
+) -> Set[tuple]:
+    """Parse and collect genes from one or more panelApp panels."""
+
+    genes = set()
+    ensembl_id_to_hgnc_id_map: Dict[str, int] = adapter.ensembl_to_hgnc_id_mapping()
+    hgnc_symbol_to_ensembl_id_map: Dict[int, str] = adapter.hgnc_symbol_ensembl_id_mapping()
+
+    with progressbar(panel_ids, label="Parsing panels", length=len(panel_ids)) as panel_ids:
+        for panel_id in panel_ids:
+            panel_dict: dict = panelapp.get_panel(panel_id)
+            panel_type_slugs = [type["slug"] for type in panel_dict.get("types")]
+            # Parse panel only if it's of the expect type(s)
+            if not set(types_filter).intersection(panel_type_slugs):
+                continue
+
+            parsed_panel = parse_panelapp_panel(
+                panel_info=panel_dict,
+                ensembl_id_to_hgnc_id_map=ensembl_id_to_hgnc_id_map,
+                hgnc_symbol_to_ensembl_id_map=hgnc_symbol_to_ensembl_id_map,
+                institute=institute,
+                confidence="green",
+            )
+            genes.update(
+                {(gene["hgnc_id"], gene["hgnc_symbol"]) for gene in parsed_panel.get("genes")}
+            )
+
+    return genes
+
+
+def load_panelapp_green_panel(adapter: MongoAdapter, institute: str, force: bool, signed_off: bool):
+    """Load/Update the panel containing all Panelapp Green genes."""
+
+    def parse_types_filter(types_filter: str, available_types: List[str]) -> List[str]:
+        """Translate panel type input from users to panel type slugs."""
+        if not types_filter:
+            return available_types
+        index_list = [int(typeint) - 1 for typeint in types_filter.replace(" ", "").split(",")]
+        return [available_types[i] for i in index_list]
+
+    # check and set panel version
+    old_panel = adapter.gene_panel(panel_id=PANEL_NAME)
+    green_panel = {
+        "panel_name": PANEL_NAME,
+        "display_name": "PanelApp Green Genes",
+        "institute": institute,
+        "version": float(math.floor(old_panel["version"]) + 1) if old_panel else 1.0,
+        "date": datetime.now(),
+    }
+
+    LOG.info("Fetching all PanelApp panels")
+
+    panel_ids = panelapp.get_panel_ids(signed_off=signed_off)
+    LOG.info(f"\n\nQuery returned {len(panel_ids)} panels\n")
+    LOG.info("Panels have the following associated types:")
+    available_types: List[str] = panelapp.get_panel_types()
+    for number, type in enumerate(available_types, 1):
+        LOG.info(f"{number}: {type}")
+    types_filter: str = input(
+        "Please provide a comma-separated list of types you'd like to use to build your panel (leave blank to use all types):  "
+    )
+    types_filter: List[str] = parse_types_filter(
+        types_filter=types_filter, available_types=available_types
+    )
+    LOG.info(f"Collecting green genes from panels of type: {types_filter}")
+    green_panel["description"] = (
```

Comment:
Description will contain the specifics of the filters applied (signed off or not, and selected types), for reference. Description was blank before.